Sources/Testing/Running/Runner.Plan.swift

```diff
@@ -304,6 +344,20 @@ extension Runner.Plan {
   }
 }
 
+extension SourceLocation {
+  /// A source location for use when synthesizing suites.
+  ///
+  /// When a suite is synthesized, it does not have accurate source location, so
+  /// the value of this property is a "runtime inferred" source location which
+  /// can be used instead.
+  //
+  // FIXME: Consider either changing `Test.sourceLocation` to Optional (an
+  // API-breaking change) or modifying Runner.Plan so that it doesn't require
+  // an actual Test instance to represent synthesized parent suites, thereby
+  // avoiding the need for a source location for them.
+  fileprivate static let synthesized = SourceLocation(fileID: "<runtime-inferred>", filePath: "<runtime-inferred>", line: 0, column: 0)
```

Comment:
The malformed file ID will cause crashes.
Line 0 and column 0 aren't allowed and will also cause crashes. Use 1 for each.

"<runtime-inferred>" reads to me like the Swift runtime inferred them, not that they were synthesized at run time. Bikeshed a bit?

Reply:
> Line 0 and column 0 aren't allowed and will also cause crashes. Use 1 for each.

Hmm… why is this not crashing currently if that's the case? This property is being accessed.

> `"<runtime-inferred>"` reads to me like the Swift runtime inferred them, not that they were synthesized at run time. Bikeshed a bit?

The compiler sometimes generates code with locations that look like `<compiler-generated>:0`, and this is what inspired me to use the `<...>` notation. Any alternative wording suggestions? `"<inferred>"` ? 

I also want to draw attention to the "FIXME" comment above this, which suggests future ways we could avoid needing this property altogether.

Reply:
> Hmm… why is this not crashing currently if that's the case? This property is being accessed.

Ah, right. `didSet` isn't used for property assignments in initializers

Reply:
> Ah, right. didSet isn't used for property assignments in initializers

Indeedy.

> Any alternative wording suggestions? "<inferred>" ?

Off the top of my head:

- `<synthesized>`
- `<compiler-generated>` (i.e. just lie for consistency with other bits of the stack)
- `<library-generated>`

But the string needs to be a valid file ID anyway (file path ideally should be valid too, but that's less urgent.) _This is why I was using the source location of the nearest real test_. It wasn't just arbitrary.

Reply:
> > Ah, right. didSet isn't used for property assignments in initializers
> 
> Indeedy.

I spun off a separate PR to improve this: https://github.com/swiftlang/swift-testing/pull/817